unstructured/ingest/connector/gcs.py

```diff
@@ -1,21 +1,37 @@
+import typing as t
 from dataclasses import dataclass
 from typing import Type
 
 from unstructured.ingest.connector.fsspec import (
     FsspecDestinationConnector,
     FsspecIngestDoc,
     FsspecSourceConnector,
+    FsspecWriteConfig,
     SimpleFsspecConfig,
 )
+from unstructured.ingest.enhanced_dataclass import enhanced_field
 from unstructured.ingest.error import SourceConnectionError
+from unstructured.ingest.interfaces import AccessConfig
 from unstructured.utils import requires_dependencies
 
 
 @dataclass
-class SimpleGcsConfig(SimpleFsspecConfig):
+class GcsAccessConfig(AccessConfig):
+    token: t.Optional[str] = enhanced_field(
+        default=None, sensitive=True, overload_name="service_account_key"
```

Comment:
overload_name is nice. so user sees `--service-account-key` but we pass it into fsspec as token?